### PR TITLE
Update contracts to Threshold Network version

### PR DESCRIPTION
**Pull request by manumonti**
NuCypher Network is going to merge with Keep Network, which will result in the Threshold Network. This implies several changes to integrate the NuCypher contracts with Threshold architecture.

In order to get this integration easier, NuCypher contracts have been added to this repository, which will become the main place for them. The objective is to have a single point for the contracts, other repositories will import the contracts from here.

This contribution updates the contracts to a version that enables the integration with the Threshold Network. The contracts have been copied from repo nucypher/nucypher branch threshold-network, commit 87fe26d.

### CODE REVIEW

**Comment by cygnusv** [comment]
For simplicity, can you just rebase the TX and remove the commits that you later revert? The commit history will look cleaner that way

**Comment by manumonti** [comment]
> For simplicity, can you just rebase the PR and remove the commits that you later revert? The commit history will look cleaner that way

✅ Done!

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ build/
 reports/
 venv/
 .python-version
+bin/
```
Comment (vzotova) [inline .gitignore:10]:
```suggestion
bin/

```

**Review by vzotova — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (manumonti) [reply .gitignore]: ✅ Done!

**Review by derekpierre — APPROVED** [review]
🎸 

Is the next step to actually build and test the contracts in this repos (presumably, something brownie based), and publish them for importing elsewhere?

**Comment by cygnusv** [comment]
> 🎸
> 
> Is the next step to actually build and test the contracts in this repos (presumably, something brownie based), and publish them for importing elsewhere?

Yes, that's the plan for later

**Review by cygnusv — APPROVED** [review]

**Pull request merged**